Frends.SFTP.DownloadFiles/Frends.SFTP.DownloadFiles/Definitions/SingleFileTransfer.cs

```diff
@@ -356,33 +411,39 @@ private void TryToRemoveDestinationTempFile()
 
         try
         {
-            if (DestinationFileExists(Path.GetFileName(DestinationFileDuringTransfer)))
-            {
-                Trace(TransferState.RemoveTemporaryDestinationFile, "Removing temporary destination file {0}", DestinationFileDuringTransfer);
-                File.Delete(DestinationFileDuringTransfer);
-            }
+            SetCurrentState(TransferState.RemoveTemporaryDestinationFile, $"Removing temporary destination file {DestinationFileDuringTransfer}.");
+            File.Delete(DestinationFileDuringTransfer);
+            _logger.NotifyInformation(BatchContext, $"FILE DELETE: Temporary destination file {DestinationFileDuringTransfer} removed.");
         }
         catch (Exception ex)
         {
-            _logger.NotifyError(BatchContext, string.Format("Could not clean up temporary destination file '{0}': {1}", DestinationFileDuringTransfer, ex.Message), ex);
+            _logger.NotifyError(BatchContext, $"Could not clean up temporary destination file '{DestinationFileDuringTransfer}': {ex.Message}.", ex);
         }
     }
 
-    private void TryToRemoveSourceTempFile(string filePath)
+    private void TryToRemoveLocalTempFile(string filePath)
     {
         try
         {
-            if (FileDefinedAndExists(filePath)) Client.Delete(filePath);
+            if (FileDefinedAndExists(filePath))
+            {
+                if ((File.GetAttributes(filePath) & FileAttributes.ReadOnly) == FileAttributes.ReadOnly)
+                {
+                    File.SetAttributes(filePath, FileAttributes.Normal); // Clear flags so readonly doesn't cause any problems CO-469
```

Comment:
Can you expand a bit on the potential problems? No need for full detailed story, but just briefly. Not everyone knows and has access to CO469